From be4c21ac1f8647b2553ddca0e9013373864e2853 Mon Sep 17 00:00:00 2001
From: ~J3mbo4D~ <creamylofly@gmail.com>
Date: Tue, 10 Mar 2026 13:04:34 +0000
Subject: [PATCH] kernel: ifdef MODULE for __stack_chk_guard

Signed-off-by: ~J3mbo4D~ <creamylofly@gmail.com>
---
 kernel/ksu.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/kernel/ksu.c b/kernel/ksu.c
index 8ba96308..92f53741 100644
--- a/kernel/ksu.c
+++ b/kernel/ksu.c
@@ -22,6 +22,7 @@
 #include "ksu.h"
 #include "file_wrapper.h"
 
+#ifdef MODULE
 // workaround for A12-5.10 kernel
 // Some third-party kernel (e.g. linegaeOS) uses wrong toolchain, which supports
 // CC_HAVE_STACKPROTECTOR_SYSREG while gki's toolchain doesn't.
@@ -37,12 +38,16 @@ unsigned long __stack_chk_guard __ro_after_init
 #else
 #define NO_STACK_PROTECTOR_WORKAROUND
 #endif
+#endif //ifdef MODULE
 
 struct cred *ksu_cred;
 
+#ifdef MODULE
 NO_STACK_PROTECTOR_WORKAROUND
+#endif
 int __init kernelsu_init(void)
 {
+#ifdef MODULE
 #if defined(CONFIG_STACKPROTECTOR) && (defined(CONFIG_ARM64) && !defined(CONFIG_STACKPROTECTOR_PER_TASK))
     unsigned long canary;
 
@@ -52,7 +57,7 @@ int __init kernelsu_init(void)
     canary &= CANARY_MASK;
     __stack_chk_guard = canary;
 #endif
-
+#endif //ifdef MODULE
 #ifdef CONFIG_KSU_DEBUG
 	pr_alert("*************************************************************");
 	pr_alert("**     NOTICE NOTICE NOTICE NOTICE NOTICE NOTICE NOTICE    **");
-- 
2.52.0

